src/transport/HighPressureGasTransport.cpp

```diff
@@ -327,167 +323,579 @@ double HighPressureGasTransport::viscosity()
 {
-    // Store current molefracs and set temp molefrac of species i to 1.0:
-    vector<double> molefracs = store(i, m_thermo->nSpecies());
+    vector<double> molefracs(m_thermo->nSpecies());
+    m_thermo->getMoleFractions(&molefracs[0]);
+    vector<double> mf_temp(m_thermo->nSpecies(), 0.0);
+    mf_temp[i] = 1;
+    m_thermo->setMoleFractions(&mf_temp[0]);
 
     double vc = m_thermo->critVolume();
+
     // Restore actual molefracs:
     m_thermo->setMoleFractions(&molefracs[0]);
     return vc;
 }
 
 double HighPressureGasTransport::Zcrit_i(size_t i)
 {
-    // Store current molefracs and set temp molefrac of species i to 1.0:
-    vector<double> molefracs = store(i, m_thermo->nSpecies());
+    vector<double> molefracs(m_thermo->nSpecies());
+    m_thermo->getMoleFractions(&molefracs[0]);
+    vector<double> mf_temp(m_thermo->nSpecies(), 0.0);
+    mf_temp[i] = 1;
+    m_thermo->setMoleFractions(&mf_temp[0]);
 
     double zc = m_thermo->critCompressibility();
+
     // Restore actual molefracs:
     m_thermo->setMoleFractions(&molefracs[0]);
     return zc;
 }
 
-vector<double> HighPressureGasTransport::store(size_t i, size_t nsp)
+double HighPressureGasTransport::low_pressure_nondimensional_viscosity(double Tr,
+                                                                       double FP,
+                                                                       double FQ) {
+    double first_term = 0.807*pow(Tr,0.618) - 0.357*exp(-0.449*Tr);
+    double second_term = 0.340*exp(-4.058*Tr) + 0.018;
+    return (first_term + second_term)*FP*FQ;
+}
+
+double HighPressureGasTransport::high_pressure_nondimensional_viscosity(double Tr,
+                                                                        double Pr,
+                                                                        double FP_low,
+                                                                        double FQ_low,
+                                                                        double P_vap,
+                                                                        double P_crit)
 {
-    vector<double> molefracs(nsp);
-    m_thermo->getMoleFractions(&molefracs[0]);
-    vector<double> mf_temp(nsp, 0.0);
-    mf_temp[i] = 1;
-    m_thermo->setMoleFractions(&mf_temp[0]);
-    return molefracs;
+
+    double Z_1 = low_pressure_nondimensional_viscosity(Tr, FP_low, FQ_low); // This is η_0*ξ
+
+    double Z_2;
+    if (Tr <= 1.0) {
+        if (Pr < P_vap/P_crit) {
+            double alpha = 3.262 + 14.98*pow(Pr, 5.508);
+            double beta = 1.390 + 5.746*Pr;
+            Z_2 = 0.600 + 0.760*pow(Pr, alpha) + (0.6990*pow(Pr, beta) - 0.60) * (1-Tr);
+        } else {
+            throw CanteraError("HighPressureGasTransport::viscosity",
+                               "State is outside the limits of the Lucas model, Tr <= 1");
+        }
+    } else if ((Tr > 1.0) && (Tr < 40.0)) {
+        if ((Pr > 0.0) && (Pr <= 100.0)) {
```

Comment:
These parentheses are unnecessary
```suggestion
    } else if (Tr > 1.0 && Tr < 40.0) {
        if (Pr > 0.0 && Pr <= 100.0) {
```